src/client/reqwest.rs

```diff
@@ -1,30 +1,38 @@
 #[cfg(feature = "reqwest_blocking")]
-use ::reqwest::blocking::Client;
+use ::reqwest::blocking::{Client, Request};
 #[cfg(feature = "reqwest_async")]
-use ::reqwest::Client;
+use ::reqwest::{Client, Request};
+
 use http::header::HeaderMap;
 use url::Url;
 
 use crate::client::ClientExt;
 
 use super::*;
+use std::convert::TryFrom;
 
 #[derive(Debug, Clone)]
 pub struct ReqwestClient(pub Client);
 
 #[maybe_async::maybe_async]
 impl ClientExt for ReqwestClient {
-    async fn request(
-        &self,
-        method: Method,
-        url: Url,
-        text: &str,
-    ) -> Result<ClientResponse, ClientError> {
+    fn new<U: Into<Option<HeaderMap>>>(headers: U) -> Result<Self, ClientError> {
+        let client = Client::builder().gzip(true);
+        match headers.into() {
+            Some(h) => client.default_headers(h),
+            None => client,
+        }
+        .build()
+        .map(|c| ReqwestClient(c))
+        .map_err(|e| ClientError::HttpClient(format!("{:?}", e)))
+    }
+
+    async fn request(&self, request: http::Request<String>) -> Result<ClientResponse, ClientError> {
+        let req: Request = request.into();
```

Comment:
My bad. It's `try_into().unwrap()` instead of `into()`.

`reqwest::Request` implements a [`TryFrom<http::Request<T>> for Request`](https://docs.rs/reqwest/0.10.6/reqwest/struct.Request.html), but NOT `From<Request<T> for Request`.